ui/src/dataExplorer/components/SaveAsTaskForm.tsx

```diff
@@ -109,14 +103,58 @@ class SaveAsTaskForm extends PureComponent<Props & WithRouterProps> {
           onSubmit={this.handleSubmit}
           canSubmit={this.isFormValid}
           dismiss={dismiss}
-          tokens={tokens}
+          tokens={this.getRelevantTokens()}
           selectedToken={selectedToken}
           onTokenChange={this.handleTokenChange}
         />
       </SpinnerContainer>
     )
   }
 
+  private getRelevantTokens() {
+    const authorizations = this.props.tokens
+
+    const {draftQueries, activeQueryIndex} = this.props
+    let readBucketName = ''
+    if (draftQueries[activeQueryIndex].editMode === 'builder') {
+      readBucketName = draftQueries[activeQueryIndex].builderConfig.buckets[0]
+    } else {
+      const text = draftQueries[activeQueryIndex].text
+      const splitBucket = text.split('bucket:')
+      const splitQuotes = splitBucket[1].split('"')
+      readBucketName = splitQuotes[1]
```

Comment:
Perhaps evaluate this queries AST.  Look for the bucket argument and use that value.  This string manipulation seems brittle.

Reply:
I have created a techdebt card for this approach and will be tackling it by the end of this week. This is a great suggestion! :)